README.md
# trialdayflutter

A new Flutter project.

## Getting Started

This project is a starting point for a Flutter application.

A few resources to get you started if this is your first Flutter project:

- [Lab: Write your first Flutter app](https://flutter.dev/docs/get-started/codelab)
- [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)

For help getting started with Flutter, view our
[online documentation](https://flutter.dev/docs), which offers tutorials,
samples, guidance on mobile development, and a full API reference.


release_notes.txt
This is a test realease.

test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:trialdayflutter/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/main.dart
// import 'package:firebase_analytics/firebase_analytics.dart';
// import 'package:firebase_analytics/observer.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';

import 'package:trialdayflutter/src/home/home.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp();

  //await FirebaseCrashlytics.instance.setCrashlyticsCollectionEnabled(true);
  // FlutterError.onError = FirebaseCrashlytics.instance.recordFlutterError;
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // static FirebaseAnalytics analytics = FirebaseAnalytics();
  // static FirebaseAnalyticsObserver observer =
  //     FirebaseAnalyticsObserver(analytics: analytics);
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'CodeMagic Trail Day',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      // navigatorObservers: <NavigatorObserver>[observer],
      home: HomePage(),
    );
  }
}


lib/src/home/home.dart
import 'package:flutter/material.dart';
import 'package:trialdayflutter/src/home/login_switch_page.dart';
import 'package:trialdayflutter/src/widget/custom_clippath.dart';

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          ClipPath(
            clipper: BackgroundClipper(),
            child: Container(
              height: 280,
              decoration: BoxDecoration(
                color: Color.fromRGBO(38, 81, 158, 1),
              ),
              child: Center(
                child: Text(
                  'CodeMagic Trial',
                  style: TextStyle(
                      color: Colors.white,
                      fontSize: 35,
                      fontWeight: FontWeight.bold,
                      letterSpacing: 1.2),
                ),
              ),
            ),
          ),
          Expanded(
            child: LoginSwitchPage(),
          ),
        ],
      ),
    );
  }
}


lib/src/widget/switch_page.dart
import 'package:flutter/material.dart';
import 'package:trialdayflutter/src/widget/custom_clippath.dart';

class SwitcPages extends StatefulWidget {
  final String loginData;

  SwitcPages({this.loginData});
  @override
  _SwitcPagesState createState() => _SwitcPagesState();
}

class _SwitcPagesState extends State<SwitcPages> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          ClipPath(
            clipper: BackgroundClipper(),
            child: Container(
              height: 280,
              decoration: BoxDecoration(
                color: Color.fromRGBO(38, 81, 158, 1),
              ),
              child: Padding(
                padding: const EdgeInsets.symmetric(
                    vertical: 32.0, horizontal: 10.0),
                child: Column(
                  children: [
                    Row(
                      mainAxisAlignment: MainAxisAlignment.start,
                      children: [
                        IconButton(
                            icon: Icon(
                              Icons.arrow_back,
                              color: Colors.white,
                            ),
                            onPressed: () {
                              Navigator.pop(context);
                            }),
                      ],
                    ),
                    SizedBox(
                      height: 30,
                    ),
                    Text(
                      widget.loginData,
                      style: TextStyle(
                          color: Colors.white,
                          fontSize: 25,
                          fontWeight: FontWeight.bold,
                          letterSpacing: 1.2),
                    ),
                  ],
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}


lib/src/home/login_switch_page.dart
import 'package:flutter/material.dart';
import 'package:trialdayflutter/src/widget/switch_page.dart';

class LoginSwitchPage extends StatefulWidget {
  @override
  _LoginSwitchPageState createState() => _LoginSwitchPageState();
}

class _LoginSwitchPageState extends State<LoginSwitchPage> {
  String email = 'Login With Email';
  String phone = 'Login With Phone Number';

  @override
  Widget build(BuildContext context) {
    return Container(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          SizedBox(
            height: 40,
          ),
          Container(
            height: 50,
            width: 280,
            decoration: BoxDecoration(
              color: Color.fromRGBO(38, 81, 158, 1),
              border: Border.all(
                color: Color.fromRGBO(38, 81, 158, 1),
              ),
              borderRadius: BorderRadius.all(Radius.circular(25.0)),
            ),
            // ignore: deprecated_member_use
            child: TextButton(
              child: Text(
                'Login With Email',
                style: TextStyle(
                    fontSize: 15, color: Colors.white, letterSpacing: 1.2),
              ),
              onPressed: () {
                Navigator.of(context).push(MaterialPageRoute(
                  builder: (context) => SwitcPages(
                    loginData: email,
                  ),
                ));
              },
            ),
          ),
          SizedBox(
            height: 20,
          ),
          Container(
            height: 50,
            width: 280,
            decoration: BoxDecoration(
              color: Color.fromRGBO(38, 81, 158, 1),
              border: Border.all(
                color: Color.fromRGBO(38, 81, 158, 1),
              ),
              borderRadius: BorderRadius.all(Radius.circular(25.0)),
            ),
            child: TextButton(
              child: Text(
                'Login with Phone Number',
                style: TextStyle(
                    fontSize: 15, color: Colors.white, letterSpacing: 1.2),
              ),
              onPressed: () {
                Navigator.of(context).push(MaterialPageRoute(
                  builder: (context) => SwitcPages(
                    loginData: phone,
                  ),
                ));
              },
            ),
          ),
          SizedBox(
            height: 20,
          ),
          Padding(
            padding: const EdgeInsets.only(left: 25.0, right: 25, top: 25),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text(
                  'You dont have an account?',
                  style: TextStyle(
                    fontSize: 13,
                    letterSpacing: 1.2,
                  ),
                ),
                GestureDetector(
                  onTap: () {},
                  child: Text(
                    'SignUp',
                    style: TextStyle(
                        fontSize: 15,
                        fontWeight: FontWeight.bold,
                        color: Colors.blue,
                        letterSpacing: 1.2),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
